src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt

```diff
@@ -233,6 +236,78 @@ class AssessmentService(
     )
   }
 
+  fun reallocateAssessment(requestUser: UserEntity, userToAllocateToId: UUID, applicationId: UUID): AuthorisableActionResult<ValidatableActionResult<AssessmentEntity>> {
+    if (! requestUser.hasRole(UserRole.WORKFLOW_MANAGER)) {
+      return AuthorisableActionResult.Unauthorised()
+    }
+
+    val assigneeUserResult = userService.getUserForId(userToAllocateToId)
+
+    val assigneeUser = when (assigneeUserResult) {
+      is AuthorisableActionResult.Success -> assigneeUserResult.entity
+      else -> return AuthorisableActionResult.NotFound()
+    }
+
+    val application = applicationRepository.findByIdOrNull(applicationId)
+      ?: return AuthorisableActionResult.NotFound()
+
+    if (application !is ApprovedPremisesApplicationEntity) {
+      throw RuntimeException("Only CAS1 Applications are currently supported")
+    }
+
+    val currentAssessment = assessmentRepository.findByApplication_IdAndDeallocatedAtNull(applicationId)
+
+    if (currentAssessment.submittedAt != null) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.GeneralValidationError("A decision has already been taken on this assessment")
+      )
+    }
+
+    val requiredQualifications = getRequiredQualificationsForApprovedPremisesApplication(application)
+
+    if (! assigneeUser.hasRole(UserRole.ASSESSOR)) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.FieldValidationError(ValidationErrors().apply { this["$.userId"] = "lackingAssessorRole" })
+      )
+    }
+
+    if (! assigneeUser.hasAllQualifications(requiredQualifications)) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.FieldValidationError(ValidationErrors().apply { this["$.userId"] = "lackingQualifications" })
+      )
+    }
+
+    currentAssessment.deallocatedAt = OffsetDateTime.now()
```

Comment:
would `currentAssessment.reallocatedAt = ...` be clearer?

as I suggested earlier, It feels like "re-allocation" is what we're all about here, rather than "de-allocation"?